File: VB-Examples/07_Conversion/ToStandAloneHTML.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet
Imports System.ComponentModel
Imports System.IO
Imports System.Text

Namespace ToStandAloneHTML
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to represent an Excel workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ToCSV.xlsx")

            ' Set the IsStandAloneHtmlFile property of HTMLOptions.Default to True
            HTMLOptions.Default.IsStandAloneHtmlFile = True

            ' Specify the output filename for the HTML file
            Dim outputFile As String = "Output.html"

            ' Create a FileStream object to write the HTML file
            Dim fileStream As New FileStream(outputFile, FileMode.Create)

            ' Save the workbook as an HTML file to the FileStream using the specified format (HTML)
            workbook.SaveToStream(fileStream, FileFormat.HTML)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'view the document
            ExcelDocViewer(outputFile)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/07_Conversion/ToStandAloneHTML.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;

namespace ToStandAloneHTML
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new Workbook object
            Workbook workbook = new Workbook();

            // Load an existing Excel document from the specified path
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ToCSV.xlsx");

            // Set the HTMLOptions to create a standalone HTML file.
            HTMLOptions.Default.IsStandAloneHtmlFile=true;

            // Specify the filename for the output HTML file.
            String outputFile = "Output.html";

            // Create a FileStream object to write the HTML file.
            FileStream fileStream = new FileStream(outputFile, FileMode.Create);

            // Save the Excel document as an HTML file to the FileStream.
            workbook.SaveToStream(fileStream, FileFormat.HTML);

            // Close the FileStream.
            fileStream.Close();

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //view the document
            ExcelDocViewer(outputFile);
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
